### PR TITLE
Make tokudb to be built on MacOS X.

### PR DESCRIPTION
**Pull request by vlad-lesin**
The functions

pthread_mutex_timedlock()
pthread_rwlock_timedrdlock()
pthread_rwlock_timedwrlock()

are not implemented in MacOS X, the simplified implementation is added.

http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1120/

### CODE REVIEW

**Comment by laurynas-biveinis** [comment]
LGTM

File: storage/tokudb/tokudb_thread.h [hunk 1]
```diff
@@ -34,6 +34,58 @@ Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
...
 
+#if (defined(__MACH__) || defined(__APPLE__)) && _POSIX_TIMERS <= 0
+
+#define _x_min(a, b) ((a) < (b) ? (a) : (b))
+
+#define timed_lock_define(timed_func_name, lock_type_name, lock_func_name)     \
+inline                                                                         \
+int                                                                            \
+timed_func_name(lock_type_name *mutex,                                         \
+                     const struct timespec *abs_timeout) {                     \
```
Comment (george-lorch) [inline storage/tokudb/tokudb_thread.h:43]: This is not quite the coding style that we are following for TokuDB. It should look more like:

```
inline int timed_func_name(
    lock_type_name* mutex,
    const struct timespec* abs_timeout) {
```

File: storage/tokudb/tokudb_thread.h [hunk 2]
```diff
@@ -34,6 +34,58 @@ Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
...
+        ts.tv_sec = 0;                                                         \
+        ts.tv_nsec = (remaining.tv_sec > 0 ?                                   \
+                      sleep_step :                                             \
+                      _x_min(remaining.tv_nsec,sleep_step));                   \
+        nanosleep(&ts, &slept);                                                \
+        ts.tv_nsec -= slept.tv_nsec;                                           \
+        if (ts.tv_nsec <= remaining.tv_nsec) {                                 \
+            remaining.tv_nsec -= ts.tv_nsec;                                   \
+        }                                                                      \
+        else {                                                                 \
```
Comment (george-lorch) [inline storage/tokudb/tokudb_thread.h]: } else {

File: storage/tokudb/tokudb_thread.h [hunk 3]
```diff
@@ -34,6 +34,58 @@ Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
...
+        }                                                                      \
+    }                                                                          \
+                                                                               \
+    return pthread_rc;                                                         \
+}
+
+timed_lock_define(pthread_mutex_timedlock,
+                  pthread_mutex_t,
+                  pthread_mutex_trylock);
+
```
Comment (george-lorch) [inline storage/tokudb/tokudb_thread.h:75]:
```
timed_lock_define(
    pthread_mutex_timedlock,
    pthread_mutex_t,
    pthread_mutex_trylock);
```

**Comment by george-lorch** [comment]
So if you can touch up the style bits, this can be merged. Thanks!

**Comment by vlad-lesin** [comment]
George, the style is fixed.

**Pull request merged**